lang/test/org/partiql/lang/eval/EvaluatingCompilerExecTests.kt

```diff
@@ -0,0 +1,302 @@
+package org.partiql.lang.eval
+
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ArgumentsSource
+import org.partiql.lang.CompilerPipeline
+import org.partiql.lang.errors.ErrorCode
+import org.partiql.lang.errors.Property
+import org.partiql.lang.errors.PropertyValueMap
+import org.partiql.lang.util.ArgumentsProviderBase
+import org.partiql.lang.util.isAnyUnknown
+import org.partiql.lang.util.softAssert
+import org.partiql.lang.util.to
+
+/**
+ * Similar class to [NullPropagatingExprFunction], but will output a stored procedure related error for invalid arity.
+ */
+abstract class NullPropagatingProcedure(
```

Comment:
Created this class because `NullPropagatingExprFunction` outputs `EVALUATOR_INCORRECT_NUMBER_OF_ARGUMENTS_TO_FUNCTION_CALL` for invalid argument arity. Other alternatives could be:
- Use `NullPropagatingExprFunction`, which would output a function call-related error for invalid arity
- Change `EVALUATOR_INCORRECT_NUMBER_OF_ARGUMENTS_TO_FUNCTION_CALL` to a more general form for function calls and procedure calls
- Create a new simple stored procedure abstract class (which may have a lot of repeated code as `NullPropagatingExprFunction`)

Reply:
Hrm.  Ok.  `NULL` propagation does not actually apply to stored procedures, i.e. `NULL` values should always pass thru to the `ExprFunction` implementation.

I think what we need is a new class named something like `StoredProcedureBase` that does the arity check but not the `NULL` propagation.

Reply:
Also, go ahead and put `StoredProcedureBase` under  `lang/src/org/partiql/lang/eval` so as to make it a part of the public API.  I'd like to avoid having customers try to implement their own arity check since throwing an `EvaluationException` with the correct error code and error context is knowledge they are not likely to have.

Reply:
Deleted `NullPropagatingProcedure` and performed the arity checks in the `EvaluatingCompiler` thunk.